File: maple-engine-plugins/maple-spark/apps/maple-spark-data-calc/src/main/scala/org/xi/maple/datacalc/spark/MapleApp.scala
package org.xi.maple.datacalc.spark

import org.apache.spark.sql.SparkSession
import org.xi.maple.datacalc.spark.api.Logging
import org.xi.maple.datacalc.spark.model.{MapleArrayData, MapleData, MapleGroupData}

object MapleApp extends Logging {

  def main(args: Array[String]): Unit = {
    val argsMap = ArgsParser.getParams(args)
    val execType = argsMap.getOrElse("exec-type", "array")

    val spark = SparkSession.builder.getOrCreate()
    try {
      val config: String = if (argsMap.contains("data")) argsMap("data") else getContent(spark, argsMap("file"))
      val data: MapleData = if ("group" == execType) MapleGroupData.getData(config) else MapleArrayData.getData(config)
      val execution = new MapleExecution(spark, data, null)
      execution.execute()
    } finally {
      spark.close()
    }
  }

  private def getContent(spark: SparkSession, path: String): String = {
    val value = spark.sparkContext.wholeTextFiles(path)
    val tuples = value.collect()
    tuples(0)._2
  }
}

File: maple-engine-plugins/maple-flink/apps/maple-flink-data-calc/src/main/java/org/xi/maple/datacalc/flink/MapleApp.java
package org.xi.maple.datacalc.flink;

import org.apache.commons.cli.CommandLine;
import org.apache.flink.configuration.Configuration;
import org.apache.flink.configuration.PipelineOptionsInternal;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.table.api.EnvironmentSettings;
import org.apache.flink.table.api.TableEnvironment;
import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
import org.apache.flink.table.api.config.TableConfigOptions;
import org.xi.maple.datacalc.flink.model.MapleArrayData;
import org.xi.maple.datacalc.flink.model.MapleData;
import org.xi.maple.datacalc.flink.model.MapleGroupData;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class MapleApp {

    public static void main(String[] args) throws Exception {
        CommandLine cmd = ArgsParser.getCommandLine(args);
        cmd.getOptionValue("file");
        String execType = cmd.getOptionValue("exec-type", "array");

        String jobId = "maple-flink-data-calc";
        Configuration configuration = new Configuration();
        configuration.setString(PipelineOptionsInternal.PIPELINE_FIXED_JOB_ID, jobId);

        try (StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment(configuration)) {
            TableEnvironment tableEnv = "stream".equals(cmd.getOptionValue("mode"))
                    ? StreamTableEnvironment.create(env, EnvironmentSettings.inStreamingMode())
                    : StreamTableEnvironment.create(env, EnvironmentSettings.inBatchMode());

            Configuration tableConfiguration = tableEnv.getConfig().getConfiguration();
            tableConfiguration.setBoolean(TableConfigOptions.TABLE_DYNAMIC_TABLE_OPTIONS_ENABLED.key(), true);

            String config = cmd.hasOption("data") ? cmd.getOptionValue("data") : getContent(cmd.getOptionValue("file"));
            MapleData data = "group".equals(execType) ? MapleGroupData.getData(config) : MapleArrayData.getData(config);
            MapleExecution<MapleData> execution = new MapleExecution<>(tableEnv, data);
            execution.execute();
        }

    }

    private static String getContent(String file) throws IOException {
        return new String(Files.readAllBytes(Paths.get(file)));
    }
}
